src/MarketingPageAcceptanceTests/MarketingPageAcceptanceTests.Actions/Utils/PageAction.cs

```diff
@@ -11,7 +12,7 @@ public abstract class PageAction
     {
         internal readonly IWebDriver driver;
         internal readonly ITestOutputHelper helper;
-
+        internal RandomGenerator random;
```

Comment:
Can be initialized here too

use `internal RandomGenerator random = new RandomGenerator();`